rio_tiler/utils.py

```diff
@@ -486,3 +487,104 @@ def create_cutline(src_dst: DataSet, geometry: Dict, geometry_crs: CRS = None) -
+
+    area = 0
+    if len(xs) > 2:
+        if xs[0] == xs[-1] and ys[0] == ys[-1]:
+            length = len(xs) - 1
+        else:
+            length = len(xs)
+        for i in range(length - 1):
+            area = area + xs[i] * ys[i + 1] - ys[i] * xs[i + 1]
+
+        area = area + xs[length - 1] * ys[0] - ys[length - 1] * xs[0]
+
+    return area >= 0
+
+
+def transform_bounds(
+    src_crs: CRS,
+    dst_crs: CRS,
+    bounds: Tuple[float, float, float, float],
+    densify_pts: int = 21,
+    coordinate_width: Optional[Union[int, float]] = None,
+):
+    """
+    Transform bounds accounting for the antimeridian.
+
+    Attributes
+    ----------
+    src_crs: str, rasterio.crs.CRS
+        Source coordinate reference system.
+    dst_crs: str, rasterio.crs.CRS
+        Target coordinate reference system.
+    bounds: Tuple
+        Left longitude, Bottom latitude, Right longitude, Top latitude of bounds.
+    densify_pts: int
+        Densify of points.
+    coordinate_width: int, float, None
+        Width of target coordinate reference system.
+        Default is None, then not accounting for the antimeridian.
+
+    Returns
+    -------
+    bool Crossing antimeridian
+    """
+
+    if densify_pts < 0:
+        raise ValueError("densify parameter must be >= 0")
+
+    if src_crs == dst_crs:
+        return bounds
+
+    (left, bottom, right, top) = bounds
+
+    in_xs = []
+    in_ys = []
+
+    if densify_pts > 0:
+        densify_factor = 1.0 / float(densify_pts + 1)
+        # left_bottom to right_bottom
+        in_xs.extend(
+            left
+            + numpy.arange(0, densify_pts + 1, dtype=numpy.float64)
+            * ((right - left) * densify_factor)
+        )
+        in_ys.extend([bottom] * (densify_pts + 1))
+        # right_bottom to right_top
+        in_xs.extend([right] * (densify_pts + 1))
+        in_ys.extend(
+            bottom
+            + numpy.arange(0, densify_pts + 1, dtype=numpy.float64)
+            * ((top - bottom) * densify_factor)
+        )
+        # right_top to left_top
+        in_xs.extend(
+            right
+            + numpy.arange(0, densify_pts + 1, dtype=numpy.float64)
+            * ((left - right) * densify_factor)
+        )
+        in_ys.extend([top] * (densify_pts + 1))
+        # left_top to left_bottom
+        in_xs.extend([left] * (densify_pts + 1))
+        in_ys.extend(
+            top
+            + numpy.arange(0, densify_pts + 1, dtype=numpy.float64)
+            * ((bottom - top) * densify_factor)
+        )
+
+    else:
+        in_xs = numpy.array(left, right, right, left, left)
+        in_ys = numpy.array(bottom, bottom, top, top, bottom)
+
+    xs, ys = transform(src_crs, dst_crs, in_xs, in_ys)
+    if coordinate_width is not None and not _is_clockwise(xs, ys):
+        xs = [x + coordinate_width if x < 0 else x for x in xs]
+        return (min(xs), min(ys), max(xs) - coordinate_width, max(ys))
```

Comment:
Is this the only difference with rasterio? if yes I feel we should ask Sean why not add this in rasterio directly!

Reply:
https://github.com/mapbox/rasterio/blob/master/rasterio/warp.py#L159
In rasterio, not account for a direction of in_xs and in_ys's rotation.
So, can't decide if it's clockwise or not.